external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -2302,19 +2443,13 @@ fn parse_struct_decl(
         visibility,
         entry,
         native,
+        macro_,
     } = modifiers;
 
     check_struct_visibility(visibility, context);
 
-    if let Some(loc) = entry {
-        let msg = format!(
-            "Invalid constant declaration. '{}' is used only on functions",
-            ENTRY_MODIFIER
-        );
-        context
-            .env
-            .add_diag(diag!(Syntax::InvalidModifier, (loc, msg)));
-    }
+    check_no_modifier(context, ENTRY_MODIFIER, entry, "struct");
+    check_no_modifier(context, MACRO_MODIFIER, macro_, "struct");
```

Comment:
Not in this PR, but as a note to myself it might be nice to come through and write a little system that knows what you're parsing and what modifiers are allowed so that we can just call `check_modifiers` in each case in these nonterminal parse functions.